pkg/logger/logger.go

```diff
@@ -13,45 +15,132 @@ import (
 
 const (
 	logTmFmtWithMS = "2006-01-02 15:04:05.000"
+	// Log level prefixes for message filtering
+	warnPrefix  = "+warn"
+	debugPrefix = "+debug"
 	// ANSI escape codes for coloring
 	colorReset  = "\033[0m"
+	colorRed    = "\033[31m"
+	colorGreen  = "\033[32m"
 	colorYellow = "\033[33m"
+	colorOrange = "\033[38;5;208m"
+	colorBlue   = "\033[34m"
+	colorPurple = "\033[35m"
 )
 
+// CustomLogger is a wrapper around logr.Logger that provides more intuitive log level methods
+type CustomLogger struct {
+	logger logr.Logger
+}
+
+// Error logs an error message with the given error and key-value pairs
+func (l *CustomLogger) Error(err error, msg string, keysAndValues ...interface{}) {
+	l.logger.Error(err, msg, keysAndValues...)
+}
+
+// Warn logs a warning message with the given key-value pairs
+func (l *CustomLogger) Warn(msg string, keysAndValues ...interface{}) {
+	// Use V(0) instead of V(1) to ensure warnings are displayed at log level 0
+	// Adds a special prefix to the message that can be detected by the FilteredCore implementation
+	l.logger.V(0).Info(warnPrefix+msg, keysAndValues...)
+}
+
+// Info logs an info message with the given key-value pairs
+func (l *CustomLogger) Info(msg string, keysAndValues ...interface{}) {
+	l.logger.Info(msg, keysAndValues...)
+}
+
+// Debug logs a debug message with the given key-value pairs
+func (l *CustomLogger) Debug(msg string, keysAndValues ...interface{}) {
+	// Don't know why l.logger.V(-1).Info is not working as Debug, hack it
+	l.logger.V(-1).Info(debugPrefix+msg, keysAndValues...)
+}
```

Comment:
I feel a little bit confused about the verbose level here. Based on the godoc, it seems Warn/Info/Debug actually calls the same log level and the only difference is whether we add warnPrefix/debugPrefix. Is this intended? If so why we still add V(x) for Warn/Debug?

```
func (l logr.Logger) V(level int) logr.Logger
V returns a new Logger instance for a specific verbosity level, relative to this Logger. In other words, V-levels are additive. A higher verbosity level means a log message is less important. Negative V-levels are treated as 0.
```

I was thinking about using
log.Info for warning
log.Info.V(1) for Info
log.Info.V(2) for debug